### PR TITLE
Limit sanity checks by Python 3.6+

### PR DESCRIPTION
**Pull request by tkrishtop**
Limit ansible-test checks to Python versions from 3.6 to 3.9 available from the ansible image.

### CODE REVIEW

**Comment by tkrishtop** [comment]
We could pin the Python version using --python argument:
```
ansible-test sanity --verbose --docker --python 3.6 --color --coverage --failure-ok
```

I'm currently trying to go through several Python versions by using a loop, something like 
```
for version in 3.6 3.7 3.8 3.9 ; do
    ansible-test sanity --verbose --docker --python $version --color --coverage --failure-ok
done
```

**Comment by tkrishtop** [comment]
I managed to limit `ansible-test` checks by a list of pre-defined Python versions. I used 3.6+ versions available in the docker image: from 3.6 to 3.9 included.

The checks `stable-2.10` and `stable-2.15` are now failing because of the comparison logic between `branch` and `main`. It should be trimmed for the output of a `for` loop I introduced.

**Comment by tkrishtop** [comment]
The checks stable-2.10 and stable-2.15 are now green but I wonder how to fix ansible-lint score. Do I need to add a changelog even for the .gihthub/workflow changes? @tonyskapunk could you please have a look?

**Comment by fredericlepied** [comment]
> The checks stable-2.10 and stable-2.15 are now green but I wonder how to fix ansible-lint score. Do I need to add a changelog even for the .gihthub/workflow changes? @tonyskapunk could you please have a look?

you can try to add `jmespath` line 27.

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -81,10 +81,13 @@ jobs:
           set -x
           git checkout -b branch
           git fetch --unshallow origin main
-          cmd="ansible-test sanity --verbose --docker --color --coverage --failure-ok"
-          $cmd 2>&1 | tee branch.output
+          for version in 3.6 3.7 3.8 3.9 ; do
+            ansible-test sanity --verbose --docker --python $version --color --coverage --failure-ok
+          done 2>&1 | tee branch.output
```
Comment (tonyskapunk) [inline .github/workflows/pr.yml:88]: Only 3.9 is being kept in the output, you want to append instead
```suggestion
          done 2>&1 | tee -a branch.output
```

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -81,10 +81,13 @@ jobs:
...
           git checkout main
-          $cmd > main.output 2>&1
+          for version in 3.6 3.7 3.8 3.9 ; do
+            ansible-test sanity --verbose --docker --python $version --color --coverage --failure-ok
+          done > main.output 2>&1
```
Comment (tonyskapunk) [inline .github/workflows/pr.yml:92]: likewise, rather append:
```suggestion
          done >> main.output 2>&1
```

**Comment by tkrishtop** [comment]
> > The checks stable-2.10 and stable-2.15 are now green but I wonder how to fix ansible-lint score. Do I need to add a changelog even for the .gihthub/workflow changes? @tonyskapunk could you please have a look?
> 
> you can try to add `jmespath` line 27.

thanks @fredericlepied it made things better.
The current error is because the filepath is not considered in the diff and the same errors but from different files are somehow messed up. I'm checking how to fix the regex.

```
## Improvements over main branch:
+FIXED: roles/vendors/kvm/tasks/exists.yml:1: warning[outdated-tag]: Replaced outdated tag 'unnamed-task' with 'name', replace it to avoid future errors
+FIXED: roles/vendors/supermicro/tasks/iso.yml:31: warning[outdated-tag]: Replaced outdated tag 'var-spacing' with 'jinja', replace it to avoid future errors
## Regressions from main branch:
-ERROR: roles/vendors/hpe/tasks/iso.yml:10: warning[outdated-tag]: Replaced outdated tag 'unnamed-task' with 'name', replace it to avoid future errors
-ERROR: roles/vendors/supermicro/tasks/disk.yml:56: warning[outdated-tag]: Replaced outdated tag 'var-spacing' with 'jinja', replace it to avoid future errors
> Fix regressions listed above
```

**Review by fredericlepied — APPROVED** [review]

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -24,12 +24,14 @@ jobs:
...
       - name: Install requirements for plugins
-        run: pip install junitparser junit_xml
+        run: pip install junitparser junit_xml jmespath
 
       - name: Compare current branch vs main branch
         # Run ansible-lint on this branch then compare vs main branch
         run: |
           set -x
+          # fix randomness
+          export PYTHONHASHSEED=42
```
Comment (tkrishtop) [inline .github/workflows/pr.yml:34]: wow, what a fix! and it works!

**Review by tonyskapunk — COMMENTED** [review]
Oops it merged too quick, it needs some rework otherwise we'll hide some results

File: .github/workflows/pr.yml (hunk 2 above)
Comment (fredericlepied) [reply .github/workflows/pr.yml:92]: started a fix for that at https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/102

**Pull request merged**